### ISSUE TITLE
[ocis] support config via .env files

### ISSUE THREAD

**Issue by micbar** [owncloud/product#16]
## What?

Define a global `.env` file that always gets merged to the environment.

## How?

Like Docker, we want to be able to load all environment vars from a file named `.env`.

For mode details on Docker's .env convention, [docs](https://docs.docker.com/compose/env-file/)

**Comment by micbar** [comment]
Viper issue https://github.com/spf13/viper/issues/455

**Comment by micbar** [comment]
@butonic @refs 
I don't agree with closing this issue.

We cannot close if we only fix it in one extension.

**Comment by refs** [comment]
indeed, it was closed since I referenced it on the PR :). This should only be closed once all extensions have the changes.

**Comment by micbar** [comment]
Can you describe the current POC in ocis-accounts?

We need to decide, if we want to go that way.

does it work with a single .env file?

**Comment by refs** [comment]
@micbar there's a description on the original PR: https://github.com/owncloud/ocis-accounts/pull/21#issuecomment-615072957

> We need to decide, if we want to go that way.

What do you mean by that? We currently support every config mode.

> does it work with a single .env file?

Please explain :)

**Comment by micbar** [comment]
The request from the client team was, if we support writing a bunch ov ENV Variables in a single .env file like docker supports it.

**Comment by refs** [comment]
Yes. That is explained in the PR :) However the only conventions are:

- file location (it MUST be a path known by Viper)
- file name (in the accounts service name, accounts.env)

Inside the file you can configure any environment variable as you would do in the CLI or Docker. Should I walk you through on how that works on a call @micbar?

**Comment by micbar** [comment]
> file name (in the accounts service name, accounts.env)

That is not, what we meant ;-)

**Comment by SolidSly** [comment]
Greetings @refs , I think what @micbar means is that it is indeed great to be able to read files that have ".env" has an extension; but it is currently not possible to read dofiles (hidden files) such as .env or .envrc.
Correct me if I didn't get it right
